stablehlo/transforms/StablehloCanonicalizeDynamism.cpp

```diff
@@ -181,6 +195,9 @@ struct CanonicalizeDynamicReshapeOpPattern
                                 PatternRewriter& rewriter) const override {
     // This pattern ignores and discards the output_shape operand. We rely on
     // the verifier to make sure that its value is consistent with result type.
+    SmallVector<int64_t> outputShape;
+    if (!succeeded(hlo::matchInts(op.getOutputShape(), outputShape)))
```

Comment:
If this pattern of unused vector for `matchInts` grows in usage (3 times in this PR), we could consider making a `matchInts` overload that takes a single argument and returns a LogicalResult, that way this vector doesn't have to stick around for the lifetime of this function call